Visual Studio 2019 compilation problems

After upgrading to the latest VS2019 and Windows SDK (as well as pulling OE master), I now get lots of instances of this:

```
[39/2560] Building C object tests\syscall\tool\CMakeFiles\rmdir.dir\rmdir_win32.c.obj
FAILED: tests/syscall/tool/CMakeFiles/rmdir.dir/rmdir_win32.c.obj
C:\PROGRA~2\MICROS~1\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x64\cl.exe  /nologo   /DWIN32 /D_WINDOWS /wd4566 /wd4200 /MDd /Zi /Ob0 /Od /RTC1 /WX /W3 -std:c11 /showIncludes /Fotests\syscall\tool\CMakeFiles\rmdir.dir\rmdir_win32.c.obj /Fdtests\syscall\tool\CMakeFiles\rmdir.dir\rmdir.pdb /FS -c ..\tests\syscall\tool\rmdir_win32.c
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9461): error C2220: the following warning is treated as an error
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9461): warning C5105: macro expansion producing 'defined' has undefined behavior
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9461): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
```

The warning is correct in that the offending macro does indeed contain a `defined`.

I'm never sure that the various bits of my Windows setup are in sync. Could someone verify that they suffer from the same problem? If so, I suppose we should disable that warning or figure out the right `#defines`  and/or compiler switches to add.

--- a/cmake/compiler_settings.cmake
+++ b/cmake/compiler_settings.cmake
@@ -122,7 +122,10 @@ elseif (MSVC)
   # so this behavior can be removed if/when cmake_minimum_required is raised to 3.15.
   # ======= Default compiler flags for cmake version 3.12 can be found here: =======
   # https://github.com/Kitware/CMake/blob/v3.12.0/Modules/Platform/Windows-MSVC.cmake
-  set(CMAKE_C_FLAGS "/DWIN32 /D_WINDOWS")
+  # With std:c11, the preprocessor in latest MSVC is more standards compliant and emits
+  # warnings in winbase.h. As recommended by the warning, /Wv:18 is added to suppress
+  # it.
+  set(CMAKE_C_FLAGS "/DWIN32 /D_WINDOWS /Wv:18")
   set(CMAKE_C_FLAGS_DEBUG "/MDd /Zi /Ob0 /Od /RTC1")
   set(CMAKE_C_FLAGS_RELEASE "/MD /O2 /Ob2 /DNDEBUG")
 

--- a/scripts/clangw
+++ b/scripts/clangw
@@ -48,6 +48,7 @@ function call_clang {
         [ "$a" == "/W3" ]           && continue
         [ "$a" == "/W4" ]           && continue
         [ "$a" == "/WX" ]           && continue
+        [ "$a" == "/Wv:18" ]        && continue
 
         # Ignore warnings for specific error codes
         if [[ "$a" =~ /[wW][dD][0-9]* ]]; then

--- a/tests/abi/host/host.cpp
+++ b/tests/abi/host/host.cpp
@@ -3,12 +3,12 @@
 
 #include <host/ecall_ids.h>
 #include <host/hostthread.h>
+#include <math.h>
 #include <openenclave/host.h>
 #include <openenclave/internal/calls.h>
 #include <openenclave/internal/error.h>
 #include <openenclave/internal/raise.h>
 #include <openenclave/internal/tests.h>
-#include <cmath>
 #include "../abi_utils.h"
 #include "abi_u.h"
 
@@ -239,7 +239,7 @@ void test_mmx_abi_poison(oe_enclave_t* enclave)
     OE_TEST(enclave_add_float(enclave, &float_result) == OE_OK);
 
     printf("x87 FPU result = %f\n", float_result);
-    OE_TEST(!std::isnan(float_result));
+    OE_TEST(!isnan(float_result));
 }
 
 void test_fpu_stack_overflow(oe_enclave_t* enclave)
@@ -266,7 +266,7 @@ void test_fpu_stack_overflow(oe_enclave_t* enclave)
     OE_TEST(enclave_add_float(enclave, &float_result) == OE_OK);
 
     printf("x87 FPU result = %f\n", float_result);
-    OE_TEST(!std::isnan(float_result));
+    OE_TEST(!isnan(float_result));
 }
 
 int main(int argc, const char* argv[])

